File: soske/comms/set_test.go
package comms

import (
	"testing"

	"github.com/stretchr/testify/assert"
	"github.com/wirehaiku/soske/soske/tools/data"
	"github.com/wirehaiku/soske/soske/tools/test"
)

func TestSetCommand(t *testing.T) {
	// setup
	db := test.DB()

	// success - new key
	outs, err := SetCommand(db, []string{"test", "one", "two"})
	body, _ := data.GetString(db, "select body from Vals where key='test' order by init, oid desc")
	assert.Empty(t, outs)
	assert.NoError(t, err)
	assert.Equal(t, "one\ntwo", body)

	// success - existing key
	outs, err = SetCommand(db, []string{"test", "three"})
	body, _ = data.GetString(db, "select body from Vals where key='test' order by init, oid desc")
	assert.Empty(t, outs)
	assert.NoError(t, err)
	assert.Equal(t, "three", body)
}

File: soske/comms/set__test.py
"""
Tests for 'soske.comms.set'.
"""

from soske.comms import set_
from soske.tools.test import conn, run


def test_set_(conn, run):
    # success - new key
    code, echo = run(conn, "set", "test", "one", "two")
    row = conn.execute("select * from GoodVals where key='test'").fetchone()
    assert code == 0
    assert echo == ""
    assert row["body"] == "one\ntwo"

    # success - existing key
    code, echo = run(conn, "set", "test", "three", "four")
    row = conn.execute("select * from GoodVals where key='test'").fetchone()
    assert code == 0
    assert echo == ""
    assert row["body"] == "three\nfour"